### DISCUSSION TITLE
Watch changes rooms frequently while i stay still

### DISCUSSION THREAD

**Discussion by alexbell62** [mKeRix/room-assistant#346] [Q&A]
Hi,
I'm just starting using room assistant so forgive me if this was asked before or obvious.

I have 4 raspberry pi zero w and lights on automation in 4 different rooms, its tracking Fitbit watches on my wrist. 
I was in the bedroom yesterday lying still but then lights started turning on in the living room and kitchen and bedroom randomly. Had to switch off automation in order to get some sleep.

Looked today in the home assistant logs and it seems the room was 'bedroom' most of the time but every few minutes it would jump to 'living room' or 'kitchen' briefly and then back to 'bedroom' again.

Here is my local.yml it is the same on all raspberry pi zero w.

```
global:
  integrations:
    - homeAssistant
    - bluetoothLowEnergy
homeAssistant:
  mqttUrl: 'mqtt://host:1883'
  mqttOptions:
    username: mqttuser
    password: mqttpassw
bluetoothLowEnergy:
  hciDeviceId: 0
  whitelist:
    - c2a26b1232d9
```

I wonder if this is known behaviour and if so how to fix that? Is having more of raspberry zero w with room assistant on them in other rooms going to fix that or make it worse?
If there is no simple way to fix it in the room assistant what are other users do as a workaround in the home assistant?

Any help appreciated! Not sure what to do now with all this.

In case it makes any difference, my layout is as this:
- raspberry pi zero w is about 1-2 meters from the bed
- kitchen raspberry pi zero w is right below the bedroom on the ground floor
- living room raspberry pi zero w  is under the stairs and a bit away from both the kitchen and bedroom (so I'm not sure why it showing me in the living room at all).
- there is another room: 'office' that is also further away from the bedroom and yesterday while i was in bedroom it never 'jumped' to the office.

Thanks.
Alex

**Comment by zigt1301** [comment]
I have similar issues with my BT-classic tracking. So far I have not been able to fix it.

One tip I can give you is to try changing the [Rolling Average](https://www.room-assistant.io/guide/entities.html#settings) of the entities that you are tracking. By turning that average up it should be possible to get a more reliable location output, though it might be a delayed output since you have to be in a location for at least half the time of the rolling average.

Another option is to lower the [maxDistance setting of your BLE tracking integration](https://www.room-assistant.io/integrations/bluetooth-low-energy.html#settings). I'm currently trying something similar with BT-classic by lowering the minRSSI value of the entities to be tracked. But again, I'm still not sure if it's working properly.
**Update 19-11-20**: By increasing the minRSSI to at least -4 (so between -4 and -1) the state measurements have become way more reliable.

Maybe another user can provide some more information on fine tuning multiple RA instances when they are close enough together and entities keep switching between them needlessly.

**Comment by alexbell62** [comment]
hi @zigt1301
thanks for the tips. Im going to try this:
```
  updateFrequency: 1
  maxDistance: 5
```

But also keen to try rolling average with 2 sec window. I only not clear how to configure it. The docs say need to specify entity id but im not clear where to take them from? 
```
Config Key: entities
Each entity that room-assistant exposes is managed by a central registry and has a locally unique ID.
```

Example from the docs:
```
entities:
  behaviors:
    d6t_occupancy_count:
      debounce:
        wait: 0.75
        maxWait: 2
```

Where do i take that entity id from (in case of example its 'd6t_occupancy_count')

Thanks,
Alex

**Reply by zigt1301** [reply]
I asked about finding the entity ID's on here a few days ago. Here is the link to the answer: [Entities configuration integrations Config Keys Discussion Post](https://github.com/mKeRix/room-assistant/discussions/342#discussioncomment-128329)

**Comment by alexbell62** [comment]
Thanks @zigt1301,  
Going to test this tonight 👍

**Comment by alexbell62** [comment]
so i tried increasing the window to 3 sec, this helped with room jumping and it works fine with not turning lights off when someone in the room.
But it doesn't work that well when need to turn it on..
Another issue im having is that it goues into 'not_home' mode very often.
 
I wonder if there are any workarounds to fix these two issues? Perhaps some more settings, etc?

Thanks,
Alex

**Reply by zigt1301** [reply]
> so i tried increasing the window to 3 sec, this helped with room jumping and it works fine with not turning lights off when someone in the room.
But it doesn't work that well when need to turn it on..

That behaviour is different than what I'm seeing. Both my RA instances have the rolling average window set to 300 for all BT-classic devices but the state does change to the 'new' room within 20 seconds. I have the rolling average set up as such:
`entities:
  behaviors:
    bluetooth-classic-a4-6c-f1-56-a2-a5-tracker:
      debounce:
        wait: 1
        maxWait: 10
      rollingAverage:
        window: 300`
I think my minRSSI at around -2 has helped with the room jumping as well but it's still 98% accurate in my estimation. 

>  Another issue im having is that it goes into 'not_home' mode very often.

Here is some information on dealing with [random incorrect not_home states](bluetooth-classic-a4-6c-f1-56-a2-a5-tracker).
I think that means that none of your RA instances can detect the device at that time. Try lowering the minRSSI maybe?

**Comment by alexbell62** [comment]
hi @zigt1301
thanks! I use BLE and the distance settings that is reverse of minrssi on Bluetooth classic i think?
btw what is `debounce:` in your settings? cant find it in the docs..

**Reply by zigt1301** [reply]
Debounce is for making a sensor wait before a state change: [Room-Assistant Entities Behaviors](https://www.room-assistant.io/guide/entities.html#behaviors)

**Reply by alexbell62** [reply]
Thanks! Will play with this setting!

**Comment by alexbell62** [comment]
It seems i have incorrect `not_home` status when the phone this watch is paired to has bluetooth on (even if watch is not shown connected to it).
